src/chainstate/stacks/db/blocks.rs

```diff
@@ -4031,14 +4036,17 @@ impl StacksChainState {
         let mut fees = 0u128;
         let mut burns = 0u128;
         let mut receipts = vec![];
+        let mut tx_index: u32 = 0;
         for tx in block.txs.iter() {
-            let (tx_fee, tx_receipt) =
+            let (tx_fee, mut tx_receipt) =
                 StacksChainState::process_transaction(clarity_tx, tx, false)?;
             fees = fees.checked_add(tx_fee as u128).expect("Fee overflow");
+            tx_receipt.tx_index = tx_index;
             burns = burns
                 .checked_add(tx_receipt.stx_burned as u128)
                 .expect("Burns overflow");
             receipts.push(tx_receipt);
+            tx_index += 1;
```

Comment:
Style nit: instead of having a distinct `mut tx_index` counter, this code should use `for (tx_index, tx) in blocks.txs.iter().enumerate()`

Reply:
Same goes for `process_microblock_transactions` above.